lib/Runtime/Library/JsBuiltInEngineInterfaceExtensionObject.cpp

```diff
@@ -40,24 +40,26 @@
     } \
 
 #define FUNCTIONKIND_VALUES(VALUE) \
-VALUE(Array, values, false) \
-VALUE(Array, keys, false) \
-VALUE(Array, entries, false) \
-VALUE(Array, indexOf, false) \
-VALUE(Array, filter, false) \
-VALUE(Array, flat, false) \
-VALUE(Array, flatMap, false) \
-VALUE(Array, forEach, false) \
-VALUE(Array, some, false) \
-VALUE(Array, every, false) \
-VALUE(Array, includes, false) \
-VALUE(Array, reduce, false) \
-VALUE(Object, fromEntries, true)
+VALUE(Array, values, false, Prototype) \
```

Comment:
It feels a little repetitive to have to specify both whether it's static and the property type. Ideally we could only specify the property type here, and then use some other thing to figure out whether it's static based on the type. Maybe something like this (syntax is approximate):

```c++
enum class IsPropertyTypeStatic : bool
{
  Prototype = false,
  Constructor = true,
  Object = true
};

// later, in the macro that wants staticness:
isStatic = static_cast<bool>(IsPropertyTypeStatic::#propertyType);
```

Reply:
@yullin-ms I dont think we had considered this option when we discussed this earlier. I like it.

Reply:
1a82165a5e48c87f6559f5c109d3810c62b0800e